src/app/ngrx/reducers/selectors/error.selectors.ts

```diff
@@ -1,20 +1,83 @@
 import { createSelector } from '@ngrx/store';
+import { selectRouter } from './router.selectors';
 import { selectEpisodeError } from './episode.selectors';
 import { selectPodcastError } from './podcast.selectors';
 import { selectEpisodeMetricsError } from './episode-metrics.selectors';
 import { selectPodcastMetricsError } from './podcast-metrics.selectors';
-import { errorType } from './error.type';
+import { selectPodcastRanksError } from './podcast-ranks.selectors';
+import { selectPodcastTotalsError } from './podcast-totals.selectors';
+import * as ACTIONS from '../../actions';
+import { PodcastMetricsModel } from '../podcast-metrics.reducer';
+import { EpisodeMetricsModel } from '../episode-metrics.reducer';
+import { RouterParams, METRICSTYPE_DOWNLOADS } from '../models';
 
-export const selectCatalogErrors = createSelector(selectPodcastError, selectEpisodeError, (podcastError, episodeError) => {
-  const errors = [];
-  if (podcastError) {
-    errors.push(`${errorType(podcastError.code)} error occurred while requesting list of podcasts`);
-  }
-  if (episodeError) {
-    errors.push(`${errorType(episodeError.code)} error occurred while requesting list of episodes`);
-  }
-  return errors;
-});
-export const selectMetricsErrors = createSelector(selectPodcastMetricsError, selectEpisodeMetricsError,
-  (podcasts, episodes) => podcasts.concat(episodes));
-export const selectErrors = createSelector(selectCatalogErrors, selectMetricsErrors, (catalog, metrics) => catalog.concat(metrics));
+export const select500ErrorReloadActions =
+  createSelector(selectRouter,
+    selectPodcastError,
```

Comment:
Like the explicit error handlers.  What happens with uncaught global exceptions?  Or do those always land in one of these categories?

Reply:
+1 gotcha